spec/models/saved_claim_spec.rb

```diff
@@ -0,0 +1,201 @@
+        expect(StatsD).to have_received(:increment).with('saved_claim.create', tags: ["form_id:#{saved_claim.form_id}"])
+      end
+
+      context 'if form_start_date is set' do
+        let(:form_start_date) { DateTime.now - 1 }
+
+        it 'increments the saved_claim.time-to-file metric' do
+          allow(StatsD).to receive(:measure)
+          saved_claim.form_start_date = form_start_date
+          saved_claim.save!
+
+          claim_duration = saved_claim.created_at - form_start_date
+          expect(StatsD).to have_received(:measure).with('saved_claim.time-to-file', claim_duration,
+                                                         tags: ["form_id:#{saved_claim.form_id}"])
+        end
+      end
+    end
+
+    context 'after destroy' do
+      it 'increments the saved_claim.destroy metric' do
+        saved_claim.save!
+        allow(StatsD).to receive(:increment)
+        saved_claim.destroy
+        expect(StatsD).to have_received(:increment).with('saved_claim.destroy',
+                                                         tags: ["form_id:#{saved_claim.form_id}"])
+      end
+    end
+  end
+
+  describe '#process_attachments!' do
+    let(:confirmation_code) { SecureRandom.uuid }
+    let(:form_data) { { some_key: [{ confirmationCode: confirmation_code }] }.to_json }
+
+    it 'processes attachments associated with the claim' do
+      attachment = create(:persistent_attachment, guid: confirmation_code, saved_claim: saved_claim)
+      saved_claim.process_attachments!
+
+      expect(attachment.reload.saved_claim_id).to eq(saved_claim.id)
+      expect(Lighthouse::SubmitBenefitsIntakeClaim).to have_enqueued_sidekiq_job(saved_claim.id)
+    end
+  end
+
+  describe '#confirmation_number' do
+    it 'returns the claim GUID' do
+      expect(saved_claim.confirmation_number).to eq(saved_claim.guid)
+    end
+  end
+
+  describe '#submitted_at' do
+    it 'returns the created_at' do
+      expect(saved_claim.submitted_at).to eq(saved_claim.created_at)
+    end
+  end
+
+  describe '#to_pdf' do
+    it 'calls PdfFill::Filler.fill_form' do
+      expect(PdfFill::Filler).to receive(:fill_form).with(saved_claim, nil)
+      saved_claim.to_pdf
+    end
+  end
+
+  describe '#update_form' do
+    it 'updates the form with new data' do
+      saved_claim.update_form('new_key', 'new_value')
+      expect(saved_claim.parsed_form['new_key']).to eq('new_value')
+    end
+  end
+
+  describe '#business_line' do
+    it 'returns empty string' do
+      expect(saved_claim.business_line).to eq('')
+    end
+  end
+
+  describe '#insert_notification' do
+    it 'creates a new ClaimVANotification record' do
+      saved_claim.save!
+
+      expect do
+        saved_claim.insert_notification(1)
+      end.to change(saved_claim.claim_va_notifications, :count).by(1)
+    end
+  end
+
+  describe '#va_notification?' do
+    let(:email_template_id) { 0 }
+
+    let!(:notification) do
+      ClaimVANotification.create(
```

Comment:
These looked pretty new, but I wasn't sure if I should add a factory for this model.

Reply:
🤔 maybe once there's a third instance, but with just one I'd hold off. hard to tell what to hardcode and what to cascade without a few exemplars to play with.